File: Programers/로또 최고순위 최저순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        
        int low = 0;
        int zero = 0;
        
        for(int i= 0; i < lottos.length; i++){
            if(lottos[i] == 0){
                zero++;
                continue;
            }
            for(int j=0; j < win_nums.length; j++){
                if(lottos[i] == win_nums[j]){
                    low++;
                }
            }
        }
        answer[0] = rank(low+zero);
        answer[1] = rank(low);
        
        return answer;
    }
    
    static int rank(int a){
        int aa= 0;
        if(a == 6){
            aa = 1;
        }else if(a == 5){
            aa = 2;
        }else if(a == 4){
            aa = 3;
        }else if(a == 3){
            aa = 4;
        }else if(a == 2){
            aa = 5;
        }else{
            aa = 6;
        }
        return aa;
    }
}

File: Programers/로또 최고순위 최저순위.js
function solution(lottos, win_nums) {
    var answer = [];
    
    var zero = 0;
    var ans =0;
    
    for(var i=0; i < lottos.length; i++){
        if(lottos[i] == 0){
            zero++;
            continue;
        }
        if(win_nums.includes(lottos[i])){
            ans++;
        }
    }
    
    var max = ans + zero;
    var min = ans;
    
    if(max == 6){
        max = 1;
    }else if(max == 5){
        max = 2;
    }else if(max == 4){
        max = 3;
    }else if(max == 3){
        max = 4;
    }else if(max == 2){
        max = 5;
    }else{
        max = 6;
    }
    
    if(min == 6){
        min = 1;
    }else if(min == 5){
        min = 2;
    }else if(min == 4){
        min = 3;
    }else if(min == 3){
        min = 4;
    }else if(min == 2){
        min = 5;
    }else{
        min = 6;
    }
    
    
    answer.push(max);
    answer.push(min);
    
    return answer;
}
